### DISCUSSION TITLE
Merge queue merges should not generate notifications

### DISCUSSION THREAD

**Discussion by djc** [community/community#65319] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Right now, as I understand it, notifications are generated for both of these events:

1. A PR is added to the merge queue
2. A PR is merged

For most of my uses, (2) is a non-event (assuming it's successful). It would be nice if there was a way of filtering out these notifications.

**Comment by d-suter** [comment]
Hi djc,

I completely understand your point. For many users, a successful merge is expected and does not require a notification. Unfortunately, GitHub does not currently provide a way to filter out these specific notifications.

As a workaround, you might consider using the GitHub API to create a custom notification system that filters out the events you are not interested in. You can use the GET /notifications endpoint to list all the notifications for the authenticated user, and then create a script that filters out the unwanted notifications and only displays the ones you are interested in.

I know this is not ideal, but it might be a temporary solution until GitHub provides more granular control over notifications.

I hope this helps.

**Reply by cjthom03** [reply]
Hi - How can one see these notifications that are generated for merge from the queue? Also notifications for PR's that fail in the queue?

The only thing I have found so far is Slack notifications, but it seems I have to subscribe to EVERY workflow event in order to see what is going on in the queue.

**Reply by d-suter** [reply]
Hi @cjthom03 ,

The notifications for merges from the queue, as well as for PRs that fail in the queue, are typically sent to the communication channels that are set up in your repository's settings or in the CI/CD pipeline. For example, if you have set up Slack notifications, then these events will be sent to the specified Slack channel.

However, as you have noted, the granularity of these notifications is not always ideal. If you are receiving notifications for every workflow event, it can be overwhelming and make it difficult to identify the important events.

You might consider creating a custom notification system using the GitHub API, as I suggested to djc. By using the GET /notifications endpoint, you can retrieve all notifications for the authenticated user, and then create a script to filter out the unwanted notifications and only display the ones you are interested in.

Additionally, you can use the GitHub API to retrieve the status of PRs in the merge queue and to check if any PRs have failed. You can then create custom notifications for these events.

Here is a basic example of how you could use the GitHub API to retrieve notifications and filter out the unwanted ones:

```python
import requests

def get_notifications():
    url = "https://api.github.com/notifications"
    headers = {
        "Authorization": "token YOUR_GITHUB_TOKEN",
    }

    response = requests.get(url, headers=headers)
    notifications = response.json()

    # filter out unwanted notifications
    filtered_notifications = [n for n in notifications if n['reason'] != 'merge']

    return filtered_notifications

def main():
    notifications = get_notifications()

    for n in notifications:
        print(n['subject']['title'])

if __name__ == "__main__":
    main()
```

In this example, the `get_notifications` function retrieves all notifications for the authenticated user and then filters out the ones with the reason 'merge'. You would need to customize this to fit your specific use case.

I understand that this is not ideal and requires additional work, but until GitHub provides more granular control over notifications, it might be the best solution.

I hope this helps.
